src/lib/utils/spatial-index.ts

```
@@ -0,0 +1,266 @@
+      );
+      this.boundingBox.maxLng = Math.max(
+        this.boundingBox.maxLng,
+        item.coordinates.lng
+      );
+    }
+
+    if (!this.grid.has(key)) {
+      this.grid.set(key, []);
+    }
+
+    this.grid.get(key)!.push(item);
```

Biome (1.9.4)

[error] Forbidden non-null assertion.

Unsafe fix: Replace with optional chain operator ?. This operator includes runtime checks, so it is safer than the compile-only non-null assertion operator


(lint/style/noNonNullAssertion)